x/staking/keeper/val_state_change.go

```diff
@@ -399,7 +400,8 @@ func sortNoLongerBonded(last validatorsByAddr) ([][]byte, error) {
 	noLongerBonded := make([][]byte, len(last))
 	index := 0
 
-	for valAddrStr := range last {
+	valAddrL := maps.Keys(last)
+	for _, valAddrStr := range valAddrL {
```

Comment:
This code isn't sorting the keys, so AFAICT doesn't produce behavior any different than ranging over the map directly?